### PR TITLE
Support for Stax

### PR DESCRIPTION
**Pull request by agrojean-ledger**
## Add Stax compatibility
- Update app for latest zxlib support,
- Remove obsolete nanos SDK gitmodule dependency,
- Use proper gitmodule for zxlib dependency (remove old zxlib version that was versioned as a directory).

## Update tests
- Update tests for compatibility with latest Zemu version.
- Commit golden snapshots of all supported devices (lns,lnx,lnsp,stax).
- Remove babel, yarn config (not used anymore),
- Update package.json (newer modules versions), add package-lock.json,
- Remove obsolete .gitignore files (use only root .gitignore instead),
- Remove snapshots-tmp directories.

## Add CI
- Add rule enforcer,
- Add build reusable workflow,
- Add Zemu test job for all devices.

### CODE REVIEW

File: app/src/json/json_parser.h [hunk 1]
```diff
@@ -33,6 +33,11 @@ extern "C" {
 /// Max number of accepted tokens in the JSON input
 #define MAX_NUMBER_OF_TOKENS   1536
 
+#if defined(TARGET_STAX)
+#undef MAX_NUMBER_OF_TOKENS
+#define MAX_NUMBER_OF_TOKENS    1024
```
Comment (agrojean-ledger) [inline app/src/json/json_parser.h:38]: Required otherwise .bss does not fit in RAM.

File: app/src/tx_display.c [hunk 2]
```diff
@@ -389,6 +389,7 @@ parser_error_t tx_display_query(uint16_t displayIdx,
     strncpy_s(outKey, get_required_root_item(root_index), outKeyLen);
 
     if (!display_cache.root_item_start_token_valid[root_index]) {
+        parser_tx_obj.query.out_val = NULL;
```
Comment (agrojean-ledger) [inline app/src/tx_display.c:394]: Maybe someone has a better fix for the dangling reference of the local `tmp_val` variable in `parser_tx_obj` ?

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    branches:
+    - master
+    - develop
+  pull_request:
+
+jobs:
+
+  guideline_enforcer:
+    name: Check guidelines
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_guidelines_enforcer.yml@fbe/fix_ICONNAME_location_when_deps_are_used
```
Comment (lpascal-ledger) [inline .github/workflows/ci-workflow.yml]: Maybe we should be able to refer to a stable version of the workflow now?

File: app/src/addr.h [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
-*   (c) 2018 Zondax GmbH
+*   (c) 2018 - 2022 Zondax AG
```
Comment (lpascal-ledger) [inline app/src/addr.h:2]: Zondax changed its company legal form?

**Review by lpascal-ledger — APPROVED** [review]
What I understand from these changes is that most of the work is done in the ZXlib, and all needing to be done on top of it is integrating the correct Makefile flags and C guards, and adapting the CI/tests for Stax, am I right?
LGTM if I am/

**Comment by agrojean-ledger** [comment]
> What I understand from these changes is that most of the work is done in the ZXlib, and all needing to be done on top of it is integrating the correct Makefile flags and C guards, and adapting the CI/tests for Stax, am I right? LGTM if I am/

Yes you are absolutely right @lpascal-ledger . The "heavy lifting" of the GUI logic is totally hidden in zondax's ZXlib. I have updated the code to work with the latest version of this lib and fix a few issues here and there but that's about all I needed to do. 

Most painful was adapting the tests, updating package versions etc etc.

File: app/src/addr.h (hunk 4 above)
Comment (agrojean-ledger) [reply app/src/addr.h:2]: It seems indeed. I copied this one from another zondax app...

File: .github/workflows/ci-workflow.yml (hunk 3 above)
Comment (agrojean-ledger) [reply .github/workflows/ci-workflow.yml]: We can probably (not definetly, need to check). I'll update that and make sure everything runs smoothly before merge.

**Pull request merged**